src/query/remote/compressed_codecs.go

```diff
@@ -169,13 +173,19 @@ func compressedSeriesFromSeriesIterator(
 			if err != nil {
 				return nil, err
 			}
-
 			replicaSegments = append(replicaSegments, segments)
 		}
 
-		compressedReplicas = append(compressedReplicas, &rpc.M3CompressedValuesReplica{
+		if reset {
+			if err := readers.Rewind(); err != nil {
```

Comment:
Honestly considering this is basically free now, may be worth just always rewinding?